pkg/brokerapi/openservicebroker/open_service_broker_client_test.go

```diff
@@ -17,22 +17,163 @@ limitations under the License.
 package openservicebroker
 
 import (
+	"net/http"
 	"testing"
 
 	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi"
+	"github.com/kubernetes-incubator/service-catalog/pkg/brokerapi/openservicebroker/util"
 )
 
-func TestUpdateServiceInstance(t *testing.T) {
-	cli := NewClient(&servicecatalog.Broker{})
+var fbs *util.FakeBrokerServer
+var fakeBroker *servicecatalog.Broker
 
-	_, err := cli.UpdateServiceInstance("foo", &brokerapi.ServiceInstanceRequest{})
-	if err == nil {
-		t.Fatalf("Expected not implemented")
+func init() {
+	fbs = &util.FakeBrokerServer{}
+	fakeBroker = &servicecatalog.Broker{
+		Spec: servicecatalog.BrokerSpec{
+			URL: "http://localhost:8000",
+		},
 	}
-	if err.Error() != "Not implemented" {
-		t.Errorf("Expected not implemented, got %v", err)
+
+	fbs.Start(8000)
```

Comment:
That was my first thought, but Go's http library doesn't seem to make it super easy to stop servers once you've started them. Will see what I can do! If you have any recommendations I'd be happy to hear them.